edge/pkg/edged/edged_pods.go

```diff
@@ -788,8 +801,17 @@ func (e *edged) makeBlockVolumes(pod *v1.Pod, container *v1.Container, podVolume
 // alter the kubelet state at all.
 func (e *edged) convertStatusToAPIStatus(pod *v1.Pod, podStatus *kubecontainer.PodStatus) *v1.PodStatus {
 	var apiPodStatus v1.PodStatus
-
-	hostIP, err := pkgutil.GetLocalIP(e.nodeName)
+	var hostIP string
+	var err error
+	var ip net.IP
+	if e.customInterfaceName != constants.DefaultCustomInterfaceName {
+		ip, err = netutil.ChooseBindAddressForInterface(e.customInterfaceName)
+		if ip != nil {
+			hostIP = ip.String()
+		}
+	} else {
+		hostIP, err = pkgutil.GetLocalIP(e.nodeName)
+	}
```

Comment:
Can you try to encapsulate a method into `pkgutil` package?
And if the `ChooseBindAddressForInterface` acquisition fails, will you consider using the `GetLocalIP ` method to attempt to obtain it?

Reply:
I can encapsulate this, sure.

And if the `ChooseBindAddressForInterface` fails, this means there is an error of some sort. In this case i would exit with the error because it is higly possible that there is a user misconfiguration e.g. wrong interface name or no IP address for that interface. 
It is better to exit in a defined way than run anyway with some sort of unknown behavior. The user expects the interface `wlan0` but gets `eth0`. As a user that would confuse me quite a bit.